src/docfx/build/toc/BuildTableOfContents.cs

```diff
@@ -36,18 +36,21 @@ public static Task<DependencyMap> Build(Context context, Document file, TableOfC
 
         public static async Task<TableOfContentsMap> BuildTocMap(IEnumerable<Document> files)
         {
-            Debug.Assert(files != null);
-
-            var builder = new TableOfContentsMapBuilder();
-            var tocFiles = files.Where(f => f.ContentType == ContentType.TableOfContents);
-            if (!tocFiles.Any())
+            using (Log.Measure("Generating TOC map"))
             {
-                return builder.Build();
-            }
+                Debug.Assert(files != null);
 
-            await ParallelUtility.ForEach(tocFiles, file => BuildTocMap(file, builder));
+                var builder = new TableOfContentsMapBuilder();
+                var tocFiles = files.Where(f => f.ContentType == ContentType.TableOfContents);
+                if (!tocFiles.Any())
+                {
+                    return builder.Build();
+                }
 
-            return builder.Build();
+                await ParallelUtility.ForEach(tocFiles, file => BuildTocMap(file, builder), Log.Progress);
```

Comment:
>Progress [](start = 96, length = 8)

what if i run buildTOC and buildXrefMap at the same time? seems the last_processed_time is a static field?